cfme/infrastructure/virtual_machines.py

```diff
@@ -1518,3 +1527,40 @@ def VIEW(self):     # noqa
 
     def step(self, *args, **kwargs):
         self.prerequisite_view.toolbar.monitoring.item_select('Utilization')
+
+
+@navigator.register(InfraVm, 'PolicySimulation')
+class PolicySimulation(CFMENavigateStep):
+    VIEW = PolicySimulationView
+    prerequisite = NavigateToSibling('Details')
+
+    def step(self, *args, **kwargs):
+        self.prerequisite_view.toolbar.policy.item_select('Policy Simulation')
+
+
+@navigator.register(InfraVmCollection, "PolicySimulation")  # noqa
+class PolicySimulation(CFMENavigateStep):  # noqa
+    VIEW = PolicySimulationView
+
+    @property
+    def navigatable(self):
+        return bool(self.obj.filters.get("name") or self.obj.filters.get("names"))
+
+    def prerequisite(self):
+        provider = self.obj.filters.get("provider")  # None if no filter
+        if provider:
+            return navigate_to(provider, "ProviderVms")
+        else:
+            return navigate_to(self.obj, "All")
+
+    def step(self, *args, **kwargs):
+        # if filter is not defined we don't want to navigate using every object
+        if not self.navigatable:
```

Comment:
I don't think I like this pattern, at least implemented at this level and not in Navmazing directly.

I'm also not convinced this is needed here: if the collection is not filtered - then I want to run policy simulation on the entire collection - the call to .all() will account for any filters.

Reply:
Agreed, I don't think it's necessary either.